cores/esp32/WString.h

```diff
@@ -35,295 +35,292 @@ class StringSumHelper;
 // an abstract class used as a means to proide a unique pointer type
 // but really has no body
 class __FlashStringHelper;
-#define F(string_literal) (reinterpret_cast<const __FlashStringHelper *>(PSTR(string_literal)))
+#define FPSTR(pstr_pointer) (reinterpret_cast<const __FlashStringHelper *>(pstr_pointer))
```

Comment:
WString.h: 38:0: warning: "FPSTR" redefined
   #define FPSTR(pstr_pointer) (reinterpret_cast<const __FlashStringHelper *>(pstr_pointer))
 
WString.h:29: In file included from
Arduino.h:152: from
[…]
pgmspace.h:35: note  this is the location of the previous definition
   #define FPSTR(p)      ((const char *)(p))

Reply:
Looks like I didn't catch that difference that needed patching, sorry.  Weird how it got through CI, though.

Did you want to change the `#define`s back in your own PR, @dok-net, or would you like me to do it?

Reply:
PR #2726 undoes the redefinition, returning it to the ESP32 native definition.  Since you have a failing case, can you check that it looks good for you?

Reply:
No more warnings with PR #2726 in place, thank you.

Reply:
merged :)